Libraries/Blob/File.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule File
+ * @flow
+ * @format
+ */
+'use strict';
+
+const Blob = require('Blob');
+
+const invariant = require('invariant');
+
+import type {BlobOptions} from 'BlobTypes';
+
+/**
+ * The File interface provides information about files.
+ */
+class File extends Blob {
+  /**
+   * Constructor for JS consumers.
+   */
+  constructor(
+    parts: Array<Blob | string>,
+    name: string,
+    options?: BlobOptions,
+  ) {
+    invariant(
+      parts != null && name != null,
+      'Failed to construct `File`: Must pass both `parts` and `name` arguments.',
+    );
+
+    super(parts, options);
+    this.data.name = name;
```

Comment:
Mutating `this.data` seems a bit like a code smell even though we know that the parent is _currently_ the only one instantiating and managing it. I don't have any suggestions, though.